[Altair] Add regression test for parsing non-Altair StandardConfig

paulhauner opened the pull request:
## Issue Addressed

NA

## Proposed Changes

~~I was wondering why `StandardConfig` was wrapped in a `SuperStruct`, so I figured I'd let the compiler tell me. I'll let this run against CI to see if there's anything I've missed.~~

Adds a regression test to ensure we can parse a `StandardConfig` without Altair params.

## Additional Info

NA

paulhauner committed:
Try remove superstruct from StandardConfig

michaelsproul commented:
This is for backwards compatibility, and I added it relatively recently. I wanted to be able to parse a "base-only" config from JSON, as this was required to work with the Pyrmont Teku nodes hosted by Infura.

With Proto's config overhaul we'll need to restructure again, so I'm against changing this for now

michaelsproul commented:
Proto's PR: https://github.com/ethereum/eth2.0-specs/pull/2390

paulhauner committed:
Revert "Try remove superstruct from StandardConfig"

This reverts commit 51c9765b063d0f7fc7fcc5f95aa6b2086f345b44.

paulhauner committed:
Add test to ensure base spec can be parsed

paulhauner commented:
> This is for backwards compatibility, and I added it relatively recently. I wanted to be able to parse a "base-only" config from JSON, as this was required to work with the Pyrmont Teku nodes hosted by Infura.

Ooh right, I didn't pick up that the `serde(untagged)` was achieving this, given the `SuperStruct` stuff in macro-land.

I've reverted my previous commit and added a regression test to prevent this happening again.

paulhauner commented:
Oh, this test already exists as `decode_no_altair`, my bad. I'll close this.

michaelsproul commented:
No worries, I appreciate the attention to detail
